### PR TITLE
EDUCATOR-5031: Delete 'Student State' for all team members

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** Reset team state when a staff member resets user state for a user on a team in a team assignment.

**What changed?**

When `clear_student_state` is called, check if this is a team assignment, and if so, call `clear_team_state`. `clear_team_state` follows the logic of `clear_user state`. We first look up the team, then the team submission, then call `team_workflow_api.cancel_workflow` and `team_submissions_api.reset_scores` ( the latter doesn't yet exist, an so this PR will fail and be unmergeable until https://github.com/edx/edx-submissions/pull/120 merges and is deployed )

**Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [x] Translations up to date
- [x] JS minified, SASS compiled
- [x] Test suites passing on Jenkins
- [x] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [EDUCATOR-5031](https://openedx.atlassian.net/browse/EDUCATOR-5031)

FIY: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/xblock/staff_area_mixin.py [hunk 1]
```diff
@@ -460,7 +460,37 @@ def clear_student_state(self, user_id, course_id, item_id, requesting_user_id):
...
+            return
+
+        if user_team is None:
+            logger.warning(
+                error_msg_base + 'but they are not on a team for course {} item {}.'.format(
+                    course_id, item_id
+            ))
+            return
+
+        from submissions import api as submission_api
```
Comment (jansenk) [inline openassessment/xblock/staff_area_mixin.py:479]: this can go

File: openassessment/xblock/staff_area_mixin.py [hunk 2]
```diff
@@ -460,7 +460,37 @@ def clear_student_state(self, user_id, course_id, item_id, requesting_user_id):
                 item_id,
                 clear_state=True
             )
-            # TODO: try to remove the above pylint disable once edx-submissions release is done
+
+    def clear_team_state(self, user_id, course_id, item_id, requesting_user_id):
+        error_msg_base = 'Attempted to clear team state for anonymous user {} '.format(user_id)
```
Comment (nsprenkle) [inline openassessment/xblock/staff_area_mixin.py:465]: I like that this builds up a human-friendly error message 👍

File: openassessment/xblock/staff_area_mixin.py [hunk 3]
```diff
@@ -524,3 +554,38 @@ def _cancel_workflow(self, submission_uuid, comments, requesting_user_id=None):
...
+            # Cancel the related workflow.
+            team_workflow_api.cancel_workflow(
+                team_submission_uuid,
+                comments,
+                requesting_user_id,
+            )
+            return {
+                "success": True,
+                'msg': self._(
+                    u"The team submission has been removed from assessment. "
```
Comment (nsprenkle) [inline openassessment/xblock/staff_area_mixin.py:579]: Nit: unnecessary `u` prefix

File: openassessment/xblock/staff_area_mixin.py [hunk 4]
```diff
@@ -460,7 +460,37 @@ def clear_student_state(self, user_id, course_id, item_id, requesting_user_id):
...
+
+        try:
+            team_submission = team_submissions_api.get_team_submission_for_team(course_id, item_id, user_team.team_id)
+        except TeamSubmissionNotFoundError:
+            logger.warning(error_msg_base + "course {} item {} but no team submission was found for team {}".format(
+                course_id, item_id, user_team.team_id
+            ))
+
+        # Remove the submission from grading pool
+        self._cancel_team_workflow(team_submission['team_submission_uuid'], "Student and team state cleared", requesting_user_id)
```
Comment (nsprenkle) [inline openassessment/xblock/staff_area_mixin.py:490]: Are these comments read/parsed anywhere? If so, should this be a defined constant or are we okay to use this ad-hoc like this?

File: openassessment/xblock/test/test_staff_area.py [hunk 5]
```diff
@@ -798,6 +798,28 @@ def test_staff_delete_student_state(self, xblock):
...
+        self.assertEqual(status_counts['teams'], 1)
+
+        xblock.clear_student_state('Bob', 'test_course', xblock.scope_ids.usage_id, STUDENT_ITEM['student_id'])
+
+        status_counts, total_submissions = xblock.get_team_workflow_status_counts()
+        self.assertEqual(total_submissions, 1)
+        status_counts = self._parse_workflow_status_counts(status_counts)
+        self.assertEqual(status_counts['cancelled'], 1)
+
+    def _parse_workflow_status_counts(self, status_counts):
```
Comment (nsprenkle) [inline openassessment/xblock/test/test_staff_area.py:817]: This 👍

File: openassessment/xblock/test/test_staff_area.py [hunk 6]
```diff
@@ -1038,6 +1060,40 @@ def _setup_xblock_and_create_submission(self, xblock, **kwargs):
...
+        xblock.teams_enabled = True
+        xblock.is_team_assignment = Mock(return_value=True)
+
+        anonymous_user_ids_for_team = ['Bob', 'Alice', 'Chris']
+        xblock.get_anonymous_user_ids_for_team = Mock(return_value=anonymous_user_ids_for_team)
+
+        # pylint: disable=protected-access
+        xblock.runtime._services['user'] = NullUserService()
+        xblock.runtime._services['user_state'] = UserStateService()
+        xblock.runtime._services['teams'] = MockTeamsService(True)  # pylint: disable=protected-access
```
Comment (nsprenkle) [inline openassessment/xblock/test/test_staff_area.py:1079]: Is this `pylint` disable redundant?

File: openassessment/xblock/test/test_staff_area.py [hunk 7]
```diff
@@ -1038,6 +1060,40 @@ def _setup_xblock_and_create_submission(self, xblock, **kwargs):
             'files_names': kwargs.get('files_names', [])
         }, ['staff'])
 
+    def _setup_team_assignment_xblock_and_create_team_submission(self, xblock, **kwargs):
+        """
+        A shortcut method to setup Team Assignment ORA xblock and add a team submission to the block.
```
Comment (nsprenkle) [inline openassessment/xblock/test/test_staff_area.py:1065]: It feels like this can be refactored to leverage more shared logic with `_setup_xblock_and_create_submission` e.g.

```
_setup_xblock_and_create_submission(self, xblock, teams_enabled=False, **kwargs):
...
```

File: openassessment/xblock/test/test_staff_area.py (hunk 6 above)
Comment (jansenk) [reply openassessment/xblock/test/test_staff_area.py:1079]: probably

File: openassessment/xblock/staff_area_mixin.py (hunk 4 above)
Comment (jansenk) [reply openassessment/xblock/staff_area_mixin.py]: It looks like this is not parsed anywhere? I couldn't find any references in code.
